core/src/main/scala/kafka/tools/ConsoleConsumer.scala

```diff
@@ -392,7 +392,7 @@ object ConsoleConsumer extends Logging {
   private[tools] class ConsumerWrapper(topic: Option[String], partitionId: Option[Int], offset: Option[Long], whitelist: Option[String],
                                        consumer: Consumer[Array[Byte], Array[Byte]], val timeoutMs: Long = Long.MaxValue) {
     consumerInit()
-    var recordIter = Collections.emptyList[ConsumerRecord[Array[Byte], Array[Byte]]]().iterator()
+    var recordIter = consumer.poll(Duration.ofMillis(0L), false).iterator
```

Comment:
It sounds like the problem is that we are counting the time before we have an assignment in the record timeout. Rather than exposing a new API, I wonder if we could use a `ConsumerRebalanceListener` which only starts the clock when we get an `onPartitionsAssigned` callback with a non-empty assignment.